backend/src/south/south-connector.ts

```diff
@@ -439,18 +502,18 @@ export default class SouthConnector<T extends SouthSettings = any, I extends Sou
       cronJob.stop();
     }
     this.cronByScanModeIds.clear();
-    this.taskJobQueue = [];
     this.logger.debug(`South connector "${this.connector.name}" (${this.connector.id}) disconnected`);
   }
 
-  async stop(): Promise<void> {
+  async stop(dataStream = true): Promise<void> {
     this.stopping = true;
-    for (const cronJob of this.cronByScanModeIds.values()) {
-      cronJob.stop();
-    }
-    this.cronByScanModeIds.clear();
     this.logger.debug(`Stopping South "${this.connector.name}" (${this.connector.id})...`);
 
+    if (dataStream) {
+      // Reload the settings only on data stream case, otherwise let the history query manage the settings
+      this.connector = this.repositoryService.southConnectorRepository.getSouthConnector(this.connector.id)!;
+    }
+
```

Comment:
There is an edgy case where this is needed: when a run is in progress, the settings can be changed by the user. If the run fails because of connection failure, it sometimes reset the connection, like in South-OPCUA.

https://github.com/OptimistikSAS/OIBus/blob/650c83a3f23ccea1ca7d363525d373d6dc6a2a42/backend/src/south/south-opcua/south-opcua.ts#L336-L338

If the settings are not reloaded, it will try to reconnect again. Hence this update here.